Do not set %{?linux_make_arch} for running make modules/modules-install

sndirsch opened the pull request.

sndirsch committed:
565.77.patch: fixes build on (open)SUSE

sndirsch commented:
@scaronni-nvidia JFYI ...

scaronni-nvidia commented:
We had the same issue here, we dropped all occurrences of `{?linux_make_arch}` in the SPEC files and the problem went away. The error in the macro is here: https://github.com/openSUSE/kernel-source/blob/master/rpm/macros.kernel-source#L27-L28

When the architecture is `x86_64`, it's remapped to `x86` that then gets passed to the kernel compilation, screwing it up: https://github.com/openSUSE/kernel-source/blob/master/rpm/macros.kernel-source#L50; as the modules do not expect that you are attempting to build on `x86`.

On top of that we had a similar issue on `aarch64`, as we had the variable `ARCH` set in the environment that gets passed down to the kernel module build when using `rpmbuild`. This resulted in `aarch64` passed to the kernel build, while in that case it expects (and normally autoedetects) `arm64`. Removing that RPM macro solved both cases.

I don't know why, but the `rpmbuild` commands in other distributions use a clean environment, in SUSE it's not cleaned and variables are imported by the environment.

sndirsch committed:
Revert "565.77.patch: fixes build on (open)SUSE"

This reverts commit ade59637e8e42a7dacd70a53b979382d893e5d14.

This needs to get fixed differently. See
https://github.com/openSUSE/nvidia-open-driver-G06/pull/11#issuecomment-2525144351

sndirsch committed:
Do not set %{?linux_make_arch} for running make modules/modules-install

Otherwise x86_64 gets redefined to x86 and build fails. See more details
here:

https://github.com/openSUSE/nvidia-open-driver-G06/pull/11#issuecomment-2525144351

sndirsch commented:
Thanks a lot. Indeed this fixes the build as well.
